repl/src/main/resources/fake_shell.py

```diff
@@ -83,6 +92,74 @@ def execute_reply_internal_error(message, exc_info=None):
     })
 
 
+class JobContextImpl(object):
+    def __init__(self):
+        self.lock = threading.Lock()
+        self.sc = global_dict['sc']
+        self.sql_ctx = None
+        self.hive_ctx = None
+        self.streaming_ctx = None
+        self.local_tmp_dir = None
+
+    def sc(self):
+        return self.sc
+
+    def sql_ctx(self):
+        if self.sql_ctx is None:
+            with self.lock:
+                if self.sql_ctx is None:
+                    self.sql_ctx = SQLContext(self.sc)
```

Comment:
`sqlContext` also exists in `global_dict`. It would be nice to use the same instance - you just need to check whether it's a `SQLContext` or a `HiveContext`, so that you don't return the wrong instance in `hive_ctx`.